### PR TITLE
Blaze Optimization: Inject site URL where necessary

### PR DESCRIPTION
**Pull request by itsmeichigo**
Related to #10951 

## Description

When reviewing #10969, @selanthiraiyan suggested injecting `Site` to `BlazeCampaignDashboardViewModel`. I checked and found that injecting complicates the code while we only need site ID and URL. 

In this PR, I updated `BlazeCampaignDashboardViewModel` and `BlazeCampaignListViewModel` with an extra param `siteURL` in the initializer, which is accessible to the view for navigation. The `BlazeWebViewModel` was also updated to accept site URL instead of site for simplicity.

Also suggested in #10969, `selectedCampaignURL` has been moved to view models of the campaign list and dashboard view for testability.

## Testing instructions

Please follow the testing steps in #10969 to ensure everything still works as before.

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by hafizrahman** [comment]
Everything works normally following instructions on https://github.com/woocommerce/woocommerce-ios/pull/10969 🥳 

Looking at code now.

File: WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListViewModel.swift [hunk 1]
```diff
@@ -100,8 +104,14 @@ final class BlazeCampaignListViewModel: ObservableObject {
         }
     }
 
-    func didSelectCampaignDetails() {
+    func didSelectCampaignDetails(_ campaign: BlazeCampaign) {
         analytics.track(event: .Blaze.blazeCampaignDetailSelected(source: .campaignList))
+
+        let path = String(format: Constants.campaignDetailsURLFormat,
+                          campaign.campaignID,
+                          siteURL.trimHTTPScheme(),
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListViewModel.swift:112]: ❓ will this still work if `siteUrl` is `""` ?

**Review by hafizrahman — APPROVED** [review]
Lots of little changes but to me they look consistent and good. I left a non-blocking question but feel free to merge after taking a look!

File: WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListViewModel.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListViewModel.swift:112]: Yes, if you look at the implementation of the method, you see it returns a URL without scheme if possible, or the initial string.

**Pull request merged**